packages/flutter/lib/src/cupertino/list_section.dart

```diff
@@ -0,0 +1,803 @@
+            Padding(
+              padding: margin,
+              child: clipBehavior == Clip.none
+                  ? decoratedChildrenGroup
+                  : ClipRRect(
+                      borderRadius: childrenGroupBorderRadius,
+                      clipBehavior: clipBehavior,
+                      child: decoratedChildrenGroup,
+                    ),
+            ),
+          if (_footer != null)
+            Align(
+              alignment: AlignmentDirectional.centerStart,
+              child: Padding(
+                padding: _type == _CupertinoListSectionType.base
+                    ? _kDefaultFooterMargin
+                    : _kInsetGroupedDefaultFooterMargin,
+                child: _footer,
+              ),
+            ),
+        ],
+      ),
+    );
+  }
+}
+
+/// An iOS-style form section.
+///
+/// The base constructor for [CupertinoFormSection] constructs an
+/// edge-to-edge style section which includes an iOS-style header, rows,
+/// the dividers between rows, and borders on top and bottom of the rows.
+///
+/// The [CupertinoFormSection.insetGrouped] constructor creates a round-edged and
+/// padded section that is commonly seen in notched-displays like iPhone X and
+/// beyond. Creates an iOS-style header, rows, and the dividers
+/// between rows. Does not create borders on top and bottom of the rows.
+///
+/// The [header] parameter sets the form section header. The section header lies
+/// above the [children] rows, with margins that match the iOS style.
+///
+/// The [footer] parameter sets the form section footer. The section footer
+/// lies below the [children] rows.
+///
+/// The [children] parameter is required and sets the list of rows shown in
+/// the section. The [children] parameter takes a list, as opposed to a more
+/// efficient builder function that lazy builds, because forms are intended to
+/// be short in row count. It is recommended that only [CupertinoFormRow] and
+/// [CupertinoTextFormFieldRow] widgets be included in the [children] list in
+/// order to retain the iOS look.
+///
+/// The [margin] parameter sets the spacing around the content area of the
+/// section encapsulating [children].
+///
+/// The [decoration] parameter sets the decoration around [children].
+/// If null, defaults to [CupertinoColors.secondarySystemGroupedBackground].
+/// If null, defaults to 10.0 circular radius when constructing with
+/// [CupertinoFormSection.insetGrouped]. Defaults to zero radius for the
+/// standard [CupertinoFormSection] constructor.
+///
+/// The [backgroundColor] parameter sets the background color behind the section.
+/// If null, defaults to [CupertinoColors.systemGroupedBackground].
+///
+/// {@macro flutter.material.Material.clipBehavior}
+///
+/// See also:
+///
+///  * [CupertinoFormRow], an iOS-style list tile, a typical child of
+///    [CupertinoFormSection].
+///  * [CupertinoListSection], an iOS-style list section.
+class CupertinoFormSection extends StatelessWidget {
```

Comment:
Just out of interest, why did you move this out of its own file? Are there private fields that need sharing or something?

Reply:
Yes, I agree. I will wager different options today and get back to you in the evening.

Reply:
Is [this](https://github.com/flutter/flutter/pull/78732/commits/f521cd31fd36c3174b7cd5537386800a2198b751) going to be OK?